airflow/providers/google/cloud/hooks/dataproc.py

```diff
@@ -281,6 +281,8 @@ def wait_for_operation(
             return operation.result(timeout=timeout, retry=result_retry)
         except Exception:
             error = operation.exception(timeout=timeout)
+            if self.check_error_for_resource_is_not_ready_msg(error.message):
+                raise GoogleAPICallError(error.message)
```

Comment:
`AirflowException` is what I try to avoid, because in the current implementation in the community the code throws and handles this exception. But this is not enough for us to handle GoogleApi specific errors.
My idea was to throw a more specific exception and then handle it on the operator side for retry. I really like your idea with the new custom Exception and I will add it with the next commit.